oml/models/projection.py

```diff
@@ -13,47 +11,22 @@
 from oml.utils.io import download_checkpoint
 
 
-def get_vit_and_mlp(
-    arch_vit: str,
-    normalise_features_vit: bool,
-    mlp_features: List[int],
-    use_multi_scale_vit: bool = False,
-) -> nn.Module:
-    """
-    Function for creation of ViT model and MLP projection.
-
-    """
-    vit = ViTExtractor(
-        weights=None,
-        arch=arch_vit,
-        normalise_features=normalise_features_vit,
-        use_multi_scale=use_multi_scale_vit,
-    )
-    mlp = MLP(vit.feat_dim, mlp_features)
-    return vit, mlp
-
-
 class ExtractorWithMLP(IExtractor, IFreezable):
     """
-    Class-wrapper for extractors which adds additional MLP (may be useful for classification losses).
+    Class-wrapper for extractors which an additional MLP.
 
     """
 
-    constructors = {
-        "vits16_224_mlp_384": partial(
-            get_vit_and_mlp,
-            arch_vit="vits16",
-            normalise_features_vit=False,
-            use_multi_scale_vit=False,
-            mlp_features=[384],
-        )
-    }
-
     pretrained_models = {
         "vits16_224_mlp_384_inshop": {
             "url": f"{STORAGE_CKPTS}/inshop/vits16_224_mlp_384_inshop.ckpt",
             "hash": "35244966",
             "fname": "vits16_224_mlp_384_inshop.ckpt",
+            "init_args": {
```

Comment:
Those are bad changes. Class attributes are calculated when you enter the script with class definition even without creating instances.

You will create VitExtractor always when you try to import anything from this file.

Reply:
<img width="304" alt="image" src="https://user-images.githubusercontent.com/43877180/224317444-89bdaab3-8220-40a5-af19-9750203be52d.png">

Reply:
I've reworked this, please check the new commit